plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/PartitionProjectionService.java

```diff
@@ -165,28 +186,55 @@ public Map<String, String> getPartitionProjectionHiveTableProperties(ConnectorTa
                             String.class::cast);
                 });
 
-        // If enabled we initialize PartitionProjection to validate properties
-        List<String> partitionColumnNames = getPartitionedBy(tableMetadata.getProperties());
-        isEnabled.ifPresent(
-                enabled -> createPartitionProjection(
-                        enabled,
-                        tableMetadata.getColumns().stream()
-                                .filter(columnMetadata -> partitionColumnNames.contains(columnMetadata.getName()))
-                                .collect(toImmutableMap(
-                                        columnMetadata -> columnMetadata.getName(),
-                                        columnMetadata -> columnMetadata.getType())),
-                        tableMetadata.getColumns()
-                                .stream()
-                                .map(ColumnMetadata::getName)
-                                .collect(toImmutableList()),
-                        metastoreTableProperties));
+        Optional<Boolean> projectionEnabledProperty = Optional.ofNullable((Boolean) tableMetadata.getProperties().get(PARTITION_PROJECTION_ENABLED));
+        if (projectionEnabledProperty.isPresent()) {
+            // If enable property in place we validate settings.
```

Comment:
assert `projectionEnabledProperty` is true here and add a comment that property would not be exposed if not enabled explicitly.